pulsar-client/src/test/java/org/apache/pulsar/client/impl/PulsarClientImplTest.java

```diff
@@ -217,4 +219,32 @@ public void testResourceCleanup() throws PulsarClientException {
             assertFalse(eventLoopGroup.isShutdown());
         }
     }
+
+    @Test
+    public void testInitializingWithExecutorProviders() throws PulsarClientException {
+        ClientConfigurationData conf = clientImpl.conf;
+        @Cleanup("shutdownNow")
+        ExecutorProvider executorProvider = new ExecutorProvider(2, "shared-executor");
+        @Cleanup
+        PulsarClientImpl client2 = PulsarClientImpl.builder().conf(conf)
+                .internalExecutorProvider(executorProvider)
+                .externalExecutorProvider(executorProvider)
+                .build();
+        @Cleanup
+        PulsarClientImpl client3 = PulsarClientImpl.builder().conf(conf)
+                .internalExecutorProvider(executorProvider)
+                .externalExecutorProvider(executorProvider)
+                .build();
+    }
+
+    @Test(expectedExceptions = IllegalStateException.class)
```

Comment:
we should add a more narrow verification of the assertion, IllegalStateException may be thrown for other reasons

Reply:
Fixed